src/main/java/com/avispl/symphony/dal/communicator/barco/clickshare/BarcoClickShareCommunicator.java

```diff
@@ -393,13 +809,64 @@ private void createFeatureControl(String name, JsonNode response, String trigger
+     * Retrieve the latest one.
+     * @return api version string (v2.0 is simplified to v2 to match the url segment)
+     * @throws UnsupportedOperationException when no available api version is retrieved
+     */
+    private String getSupportedVersion() throws Exception {
+        JsonNode response = doGet(supportedApiVersion + API_SUPPORTED_VERSIONS, JsonNode.class);
+        String version = "";
+        if(response.get("status").asInt() == 200){
+            ArrayNode versions = (ArrayNode) response.get("data").get("value");
+            version = versions.get(versions.size() - 1).asText();
+            if(version.startsWith("v2")){
+                version = "v2";
+            }
+        }
+        if(StringUtils.isNullOrEmpty(version)){
+            throw new UnsupportedOperationException("Unable to retrieve a supported ClickShare API version.");
+        }
+        return version;
+    }
+
+    /**
+     * Unpack network related data from json to statistics
+     * @param node json node that contains specific network-related info
+     * @param key key to retrieve network related properties from json
+     * @param statistics map to save statistics data to
      * */
-    private List<String> getSupportedOperations() throws Exception {
-        ArrayNode supportedOperations = doGet(OPERATIONS_SUPPORTED, ArrayNode.class);
-        return StreamSupport.stream(supportedOperations.spliterator(), false).map(JsonNode::asText).collect(Collectors.toList());
+    private void populateNetworkConfigurationData(JsonNode node, String key, Map<String, String> statistics){
+        ArrayNode networkProperties = (ArrayNode) node.get(key);
+        if(networkProperties.size() == 0){
+            return;
+        }
+        networkProperties.elements().forEachRemaining(jsonNode -> {
+            int id = jsonNode.get("id").asInt();
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Operation Mode", key, id), jsonNode.get("operationMode"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Addressing", key, id), jsonNode.get("addressing"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Status", key, id), jsonNode.get("status"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Ip Address", key, id), jsonNode.get("ipAddress"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Subnet Mask", key, id), jsonNode.get("subnetMask"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s Default Gateway", key, id), jsonNode.get("defaultGateway"));
+            addStatisticsProperty(statistics, String.format("Network configuration: %s#Configuration %s MAC Address", key, id), jsonNode.get("macAddress"));

```

Comment:
Move to a separate method with proper name

Reply:
Agree, will do